server/src/main/java/io/deephaven/server/table/validation/MethodListInvocationValidator.java

```diff
@@ -0,0 +1,186 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.server.table.validation;
+
+import io.deephaven.UncheckedDeephavenException;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+/**
+ * An invocation validator that has a hardcoded list of classes and methods to permit.
+ */
+public class MethodListInvocationValidator implements MethodInvocationValidator {
```

Comment:
Why does `AnnotationMethodInvocationValidator` cache results, but this doesn't? it seems like a caching wrapper could make sense to enclose both instances - or put it higher still in the ParsingColumnExpressionValidator, so we check the cache and compute the result if absent?

Reply:
I added a caching layer for methods/constructors.  I still included the class-level annotation caching to avoid having to read a classes's annotations repeatedly for a new method.